src/v/datalake/coordinator/state_machine.cc

```diff
@@ -129,27 +129,38 @@ ss::future<> coordinator_stm::do_apply(const model::record_batch& b) {
 
 model::offset coordinator_stm::max_collectible_offset() { return {}; }
```

Comment:
Doesn't necessarily need to be part of this PR, but should we be able to implement a meaningful max collectible offset now?

Reply:
Ah got it, so just making sure I understand, snapshotting here also removes the log segments, is that right?

Reply:
yes, writing the raft snapshot also prefix truncates the segments up until that offset.